File: backend/models/Guest.js
const mongoose = require('mongoose');

const guestSchema = new mongoose.Schema({
    fullName: { type: String, required: true },
    nationality: { type: String },
    passportNumber: { type: String },
    passportCopyUrl: { type: String }, // path to the uploaded file
    dateOfBirth: { type: Date },
    gender: { type: String, enum: ['Male', 'Female', 'Other'] },
    contactNumber: { type: String },
    email: { type: String },
    hotelBranch: {
        type: String,
        required: true,
        enum: ['Hiru Villa', 'Hiru Om', 'Hiru Mudhra', 'Hiru Aadya']
    },
    roomNumber: { type: String },
    arrivalDate: { type: Date },
    departureDate: { type: Date },
    treatmentPackage: { type: String },
    specialNotes: { type: String },
    paymentStatus: { type: String, enum: ['Pending', 'Partial', 'Paid'], default: 'Pending' },
    totalAmount: { type: Number, default: 0 },
    advancePayment: { type: Number, default: 0 },
    balance: { type: Number, default: 0 },
}, { timestamps: true });

guestSchema.pre('save', function (next) {
    this.balance = this.totalAmount - this.advancePayment;
    if (this.balance <= 0 && this.totalAmount > 0) {
        this.paymentStatus = 'Paid';
    } else if (this.advancePayment > 0) {
        this.paymentStatus = 'Partial';
    }
    next();
});

module.exports = mongoose.model('Guest', guestSchema);

File: spring-backend/src/main/java/com/hiruresort/backend/models/Guest.java
package com.hiruresort.backend.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;

@Data
@Document(collection = "guests")
public class Guest {
    @Id
    @JsonProperty("_id")
    private String id;
    
    private String fullName;
    private String nationality;
    private String passportNumber;
    private String passportCopyUrl;
    private Date dateOfBirth;
    private String gender;
    private String contactNumber;
    private String email;
    private String hotelBranch;
    private String roomNumber;
    private Date arrivalDate;
    private Date departureDate;
    private Integer pax = 1;
    private String agent;
    private String remark;
    private Date visaExpiryDate;
    private String treatmentPackage;
    private String specialNotes;
    private String paymentStatus = "Pending";
    private Double totalAmount = 0.0;
    private Double advancePayment = 0.0;
    private Double balance = 0.0;
    private Date createdAt = new Date();
    private Date updatedAt = new Date();
}
